pkg/node/node_address_linux.go

```diff
@@ -28,116 +28,141 @@ import (
 	"github.com/vishvananda/netlink"
 )
 
-// firstGlobalV4Addr returns the first IPv4 global IP of an interface, where
-// the IPs are sorted in ascending order. when intf is defined only IPs
-// belonging to that interface are considered. If preferredIP is present in the
-// IP list it is returned irrespective of the sort order. Passing intf and
-// preferredIP will only return preferredIP if it is in the IPs that belong to
-// intf. In all cases, if intf is not found all interfaces are considered.
-func firstGlobalV4Addr(intf string, preferredIP net.IP) (net.IP, error) {
+var privateIPBlocks []*net.IPNet
```

Comment:
I would add this code to another package, like `pkg/ip/` something like pkg/ip/utils.go